### PR TITLE
Domains: Add availability precheck before adding a domain suggestion to the cart

### PR DESCRIPTION
**Pull request by hambai**
Since we may get some false positives from the domain suggestions engines we should handle better domain names that are not actually available. The idea is to add availability pre-check before adding the domain to the shopping cart.

#### Changes proposed in this Pull Request

* add availability precheck before adding the domain to the cart
* disable domain suggestion card if the domain is unavalable

#### Testing instructions

* inject a domain that is already registered in the REST API endpoint result
* see what happens when you try to purchase it (via calypso/nux)
* test that subdomains still work
* everything else that I didn't think of?

see also: p8kIbR-sq-p2

### CODE REVIEW

File: client/components/domains/domain-registration-suggestion/index.jsx [hunk 1]
```diff
@@ -119,8 +132,23 @@ class DomainRegistrationSuggestion extends React.Component {
 					  } )
 					: translate( 'Select', { context: 'Domain mapping suggestion button' } );
 		}
+
+		let buttonProps = { primary: true };
+		if ( this.isUnavailableDomain( suggestion.domain_name ) ) {
+			buttonProps = { primary: true, disabled: true };
+			buttonContent = translate( 'Unavailable', {
```
Comment (a8ci18n) [inline client/components/domains/domain-registration-suggestion/index.jsx]: Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Post unavailable' )` *ES Score: 14*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/pre-check-domain-availability)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

**Comment by hambai** [comment]
@klimeryk You're right! I just added tracks events to the availability precheck, so we can actually keep an eye of how often this happens.

**Comment by aidvu** [comment]
Hmmm. What happened here. :) A rebase gone wrong?

**Comment by delputnam** [comment]
Heh... Yes. I have a PR that is based on this branch...rebased the wrong branch. I figured it was harmless since it was an empty rebase.

**Comment by hambai** [comment]
Closing in favour of https://github.com/Automattic/wp-calypso/pull/30283

**Pull request closed without merging**